src/detection/localip/localip_linux.c

```diff
@@ -192,6 +318,12 @@ const char* ffDetectLocalIps(const FFLocalIpOptions* options, FFlist* results)
                         iface->mtu = (int32_t) ifr.ifr_mtu;
                 }
 
+                if (options->showType & FF_LOCALIP_TYPE_FLAGS_BIT)
+                {
+                    if (ioctl(sockfd, SIOCGIFFLAGS, &ifr) == 0)
+                        writeNIFlagsToStrBuf(&iface->flags, ifr.ifr_flags, nicFlags);
+                }
```

Comment:
getifaddrs will return all interfaces in the local system which we already got. It might be heavier than creating a socket then calling a ioctl.
And manpages say this function is not a part of the standard, it could be different on other systems.
So, I think using ioctl and ifreq here may be enough.

Reply:
Oh, getifaddrs already called.

Reply:
Thanks for this advice, I'll change to reuse its result.

Reply:
Done.